### DISCUSSION TITLE
Menu, Background and On Battery Frame Limiters

### DISCUSSION THREAD

**Discussion by GGamer6458** [ppy/osu#22364] [Ideas] [answered]
It would be nice if there was an in-game frame limiter for when you're in the menu, when osu! is in the background and when you're on battery, so it doesn't use more resources than it needs to use.

**Comment by peppy** [comment, marked as answer]
It should not be using any resources in the background (minimised), and already runs at a reduced frame rate when not focused. Can you give us a better idea of what hardware / system you're running on and what you're seeing in terms of battery usage?

**Reply by honguyenminh** [reply]
Basically, from my understanding, he wants a frame rate limiter separately between gameplay and menu, something like valorant and counter strike have, menus capped to 60-120 while gameplay is 480 or unlimited/vsync etc.

He probably also want a "battery saver" mode where osu will cap the frame rate and reduce background tasks, but I don't think that will do much to save battery in practice, just cap the frame rate manually is enough imo.

**Reply by GGamer6458** [reply]
Yes, I was thinking of the frame limit options that games like Valorant and CS:GO has implemented, where you can set frame caps in certain scenarios, like when you're in the menu or when you're on battery.

Of course you could cap the FPS using third-party software, like RTSS or NVCPL, whether that be in the foreground or the background, but I think it would be a lot easier and probably also better if there was an option within the game itself to limit the FPS in those scenarios.

Also I think having the option to change the frame limit values in-game to any value up to 1000 FPS, just like how in Valorant it doesn't let you go past a certain threshold (like 500 or 1000 FPS max), would be nice to have alongside the current preset options.

**Reply by peppy** [reply]
Have you considered changing your system power profile at an OS level? This is going to give you much better results and create a better UX across the board. I've never seen a game offset special settings for battery power, because that's *not the responsibility of the game*.

Set windows to "power saver" and your CPU and GPU will be clocked down to save power. You could also consider using Ctrl+F7 to change the frame limiter in game as you require.

We will not be adding battery specific settings. We may add a sane frame limit at menus similar to stable, but that will come later.
